src/Microsoft.AspNet.Mvc.TagHelpers/LabelTagHelper.cs

```diff
@@ -45,10 +45,12 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
                 {
                     output.MergeAttributes(tagBuilder);
 
+                    var childContent = await context.GetChildContentAsync();
+
                     // We check for whitespace to detect scenarios such as:
                     // <label for="Name">
                     // </label>
-                    if (string.IsNullOrWhiteSpace(output.Content))
+                    if (string.IsNullOrWhiteSpace(childContent) && string.IsNullOrEmpty(output.Content))
```

Comment:
be more explicit here and set `output.Content` anytime it is `null` or empty.  don't fall into the magic "if `GetChildContentAsync` was called but `Content` wasn't set" hole.